### PR TITLE
Include market strategy in app definition

### PR DESCRIPTION
**Pull request by mfranciszkiewicz**
Currently a `DEFAULT_REQUESTOR_MARKET_STRATEGY` is enforced

### CODE REVIEW

**Review by Wiezzel — COMMENTED** [review]
Looks OK but what's the deal with thousands of changed lines in taskserver and tasksession?

**Comment by mfranciszkiewicz** [comment]
That was a small hiccup with my IDE. Fixed

File: golem/marketplace/__init__.py [hunk 1]
```diff
@@ -15,5 +17,36 @@
...
+            return cls[name]
+        except KeyError:
+            return cls.default()
+
+    @classmethod
+    def default(cls):
+        raise NotImplemented
+
+
+class RequestorMarketStrategies(NameSerializableEnum):
```
Comment (Wiezzel) [inline golem/marketplace/__init__.py]: Why plural name? We never used plural names for enums or any other classes.

File: golem/marketplace/__init__.py [hunk 2]
```diff
@@ -15,5 +17,36 @@
...
+    Brass = RequestorBrassMarketStrategy
+    Wasm = RequestorWasmMarketStrategy
```
Comment (Wiezzel) [inline golem/marketplace/__init__.py]: Our styleguide doesn't mention enums but i think the options should be either in `ALL_CAPS` or `snake_case`.

File: golem/marketplace/__init__.py [hunk 3]
```diff
@@ -15,5 +17,36 @@
...
+
+    @classmethod
+    def default(cls) -> 'RequestorMarketStrategies':
+        return cls.Brass
+
+
+class ProviderMarketStrategies(NameSerializableEnum):
```
Comment (Wiezzel) [inline golem/marketplace/__init__.py]: See comments about `RequestorMarketStrategies`.

**Review by Wiezzel — COMMENTED** [review]
Some code style suggestions.

File: golem/marketplace/__init__.py (hunk 1 above)
Comment (mfranciszkiewicz) [reply golem/marketplace/__init__.py]: `RequestorMarketStrategy` already exists. Other names I came up with weren't good enough. `RequestorMarketStrategyEnum`?

Comment (Wiezzel) [reply golem/marketplace/__init__.py]: Oh, let's leave it then. I cannot think of any better name now.

**Review by Wiezzel — APPROVED** [review]

**Comment by mfranciszkiewicz** [comment]
@Wiezzel Please re-review. One of the tests was failing but due to the impact of the bug on the rest of the codebase I've changed the solution

**Review by Wiezzel — APPROVED** [review]
OK

**Pull request merged**